Formula/hbase.rb

```diff
@@ -55,6 +61,11 @@ def install
               "export HBASE_OPTS=\"-Djava.net.preferIPv4Stack=true -XX:+UseConcMarkSweepGC\"")
       s.gsub!("# export JAVA_HOME=/usr/java/jdk1.6.0/",
               "export JAVA_HOME=\"$(/usr/libexec/java_home)\"")
+
+      # Default `$HBASE_HOME/logs` is unsuitable as it would cause writes to the
+      # formula's prefix.
+      s.gsub!(/^# export HBASE_LOG_DIR=.*$/,
+              "export HBASE_LOG_DIR=#{var}/log/hbase")
```

Comment:
I just realized this also prevents legitimate overrides of this variable, e.g. in the [test of the `hbase` formula](https://github.com/Homebrew/homebrew-core/blob/b9c7d819782ac7062273111ae7cf4d48e647b4c7/Formula/hbase.rb#L156). Will be tweaking this a bit to allow for this kind of customization.